java/vortex-spark/src/main/java/dev/vortex/spark/write/VortexDataWriter.java

```diff
@@ -203,12 +207,12 @@ private void populateVector(FieldVector vector, DataType dataType, InternalRow r
         } else if (dataType instanceof StringType) {
             UTF8String str = row.getUTF8String(fieldIndex);
             if (str != null) {
-                ((VarCharVector) vector).set(rowIndex, str.getBytes());
+                ((VarCharVector) vector).setSafe(rowIndex, str.getBytes());
```

Comment:
setSafe will allocate a larger buffer if we try to insert more strings than the initial allocation (default is 32KiB)